### DISCUSSION TITLE
How to enable EditorDialog and GridEditor in sergen in core for command line

### DISCUSSION THREAD

**Discussion by Nafhaa** [serenity-is/Serenity#5834] [Help]
How to enable EditorDialog and GridEditor in sergen in core for command line

**Comment by VictorTomaili** [comment]
There is no option for creating these types from sergen.

**Reply by Nafhaa** [reply]
@VictorTomaili  thanks for reply, is there any way to handle the editor and grideditor functionalities for the details modules in master details section

**Reply by VictorTomaili** [reply]
Check [order detail](https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.Northwind/Serenity.Demo.Northwind/OrderDetail/OrderDetailsEditor.ts) sample
https://serenity.is/demo/Northwind/Order

**Reply by Nafhaa** [reply]
I mean to auto generate these files while running sergen

**Reply by VictorTomaili** [reply]
There is no option for that. But pull requests are welcome for sergen and code generation improvements. 
An option can be added to generate a command for generating grid editor etc. 
https://github.com/serenity-is/Serenity/tree/master/src/Serenity.Net.CodeGenerator

**Reply by Nafhaa** [reply]
Thanks

**Reply by VictorTomaili** [reply]
And an alternative, you can customize templates as you wish. add option `CustomTemplates` as path inside your project ([see this](https://github.com/serenity-is/Serenity/blob/master/src/Serenity.Net.CodeGenerator/Commands/GenerateCommand.cs#L70)), and modify [these templates](https://github.com/serenity-is/Serenity/tree/master/src/Serenity.Net.CodeGenerator/Templates) for your requirements. You can add a grid editor inside the `Grid.scriban` template.

**Reply by HassanBakri** [reply]
hello everyone,
i've followed the orderdetails example and when i click on master gird(to edit ) or new button i get the following error 
`system.ts:675  Uncaught Exception: "RBMSDP.Service.TieredRevenueSharingEditor" editor class not found! `

i've attached the files here [TieredRevenueSharingModel.zip](https://github.com/serenity-is/Serenity/files/13402849/TieredRevenueSharingModel.zip)
